include/dali/core/os/shared_mem.h

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_OS_SHARED_MEM_H_
+#define DALI_CORE_OS_SHARED_MEM_H_
+
+#include <stdint.h>
+#include <memory>
+#include <string>
+#include "dali/core/common.h"
+#include "dali/core/error_handling.h"
+#include "dali/core/unique_handle.h"
+
+namespace dali {
+
+#if defined(__unix__)
+using shm_handle_t = int;
+using fd_handle_t = int;
+
+#define POSIX_CHECK_STATUS_EX(status, call_str, message)                               \
+  do {                                                                                 \
+    if (status == -1) {                                                                \
+      std::string errmsg(256, '\0');                                                   \
+      int e = errno;                                                                   \
+      strerror_r(e, &errmsg[0], errmsg.size());                                        \
+      DALI_FAIL(make_string("Call to ", call_str, " failed. ", errmsg, " ", message)); \
+    }                                                                                  \
+  } while (0)
+
+
+#define POSIX_CHECK_STATUS(status, call_str) POSIX_CHECK_STATUS_EX(status, call_str, "")
+
+#define POSIX_CALL_EX(code, message)               \
+  do {                                             \
+    int status = code;                             \
+    POSIX_CHECK_STATUS_EX(status, #code, message); \
+  } while (0)
+
+
+#define POSIX_CALL(code) POSIX_CALL_EX(code, "")
+
+#else
+#error Platform not supported
+#endif
+
+class DLL_PUBLIC FdHandle : public UniqueHandle<fd_handle_t, FdHandle> {
```

Comment:
FdHandle = file descriptor handle?
```suggestion
class DLL_PUBLIC FileHandle : public UniqueHandle<fd_handle_t, FdHandle> {
```

Reply:
I'm not fond of hiding the fact of using the fd, as this is basically used in the Unix-specific implementation, but whatever.